### PR TITLE
Support bool dtypes in `chainerx::{Max,Min}imum`

### PR DESCRIPTION
**Pull request by asi1024**
This PR makes `chainerx::Maximum` and `chainerx::Minimum` accept Boolean dtypes like as `chainerx::Greater` and `chainerx::Less` do.

### CODE REVIEW

File: chainerx_cc/chainerx/routines/logic.cc [hunk 1]
```diff
@@ -54,8 +54,12 @@ Array LogicalNot(const Array& x) {
 
 Array LogicalAnd(const Array& x1, const Array& x2) { return LogicBinary<LogicalAndKernel>(x1, x2); }
 
+Array LogicalAnd(const Array& x1, Scalar x2) { return static_cast<bool>(x2) ? x1 : Zeros(x1.shape(), Dtype::kBool, x1.device()); }
```
Comment (niboshi) [inline chainerx_cc/chainerx/routines/logic.cc]: Is this implementation correct?
It looks the returned dtype is dependent on the value of `x1`.

File: chainerx_cc/chainerx/routines/logic.cc [hunk 2]
```diff
@@ -54,8 +54,12 @@ Array LogicalNot(const Array& x) {
...
+
 Array LogicalOr(const Array& x1, const Array& x2) { return LogicBinary<LogicalOrKernel>(x1, x2); }
 
+Array LogicalOr(const Array& x1, Scalar x2) { return static_cast<bool>(x2) ? Ones(x1.shape(), Dtype::kBool, x1.device()) : x1; }
```
Comment (niboshi) [inline chainerx_cc/chainerx/routines/logic.cc]: Ditto.

File: chainerx_cc/chainerx/routines/logic.cc [hunk 3]
```diff
@@ -25,11 +25,19 @@ class LogicBinaryImpl {
     }
 };
 
+void CheckLogicDtypes(DtypeKind kind1, DtypeKind kind2) {
+    if ((kind1 == DtypeKind::kBool) != (kind2 == DtypeKind::kBool)) {
+        throw DtypeError{"Comparison does not support Boolean and another mixed dtypes"};
```
Comment (niboshi) [inline chainerx_cc/chainerx/routines/logic.cc]: This sentence does not sound natural. How about this:
"Comparison of bool and non-bool dtypes is not supported."

File: chainerx_cc/chainerx/routines/math.cc [hunk 4]
```diff
@@ -469,10 +469,20 @@ Array Sum(const Array& a, const OptionalAxes& axis, bool keepdims) {
 
 namespace {
 
+void CheckComparisonDtypes(DtypeKind kind1, DtypeKind kind2) {
+    if ((kind1 == DtypeKind::kBool) != (kind2 == DtypeKind::kBool)) {
+        throw DtypeError{"Comparison does not support Boolean and another mixed dtypes"};
```
Comment (niboshi) [inline chainerx_cc/chainerx/routines/math.cc]: Ditto.

File: chainerx_cc/chainerx/routines/logic.cc (hunk 3 above)
Comment (asi1024) [reply chainerx_cc/chainerx/routines/logic.cc]: Fixed.

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by asi1024** [comment]
Sorry, I forgot to fix `TestMinimumScalar` and `TestMaximumScalar`. I'll fix them.

**Comment by niboshi** [comment]
Jenkins, test this please

**Pull request merged**